packages/core/src/operator/operator-runtime/operator-v1/processNewChallenge.ts

```diff
@@ -0,0 +1,113 @@
+import { PoolInfo, RefereeConfig, SentryKey, SentryWallet } from "@sentry/sentry-subgraph-client";
+import { getBoostFactor as getBoostFactorRPC, getSubmissionsForChallenges, KEYS_PER_BATCH, NodeLicenseStatus, ProcessChallenge, submitMultipleAssertions } from "../../index.js";
+import { operatorState } from "../operatorState.js";
+import { updateNodeLicenseStatus } from "../updateNodeLicenseStatus.js";
+import { createAssertionHashAndCheckPayout } from "./createAssertionHashAndCheckPayout.js";
+import { calculateBoostFactor } from "./calculateBoostFactor.js";
+import { retry } from "../../../index.js";
+
+/**
+ * Processes a new challenge for all the node licenses.
+ * @param {bigint} challengeNumber - The challenge number.
+ * @param {ProcessChallenge} challenge - The challenge object.
+ * @param {Challenge} challenge - The challenge.
+ */
+export async function processNewChallenge(
+    challengeNumber: bigint,
+    challenge: ProcessChallenge,
+    nodeLicenseIds: bigint[],
+    sentryKeysMap: { [keyId: string]: SentryKey },
+    sentryWalletMap?: { [address: string]: SentryWallet },
+    mappedPools?: { [poolAddress: string]: PoolInfo },
+    refereeConfig?: RefereeConfig
+) {
+    operatorState.cachedLogger(`Processing new challenge with number: ${challengeNumber}.`);
+    operatorState.cachedBoostFactor = {};
+
+    const batchedWinnerKeys: bigint[] = []
+
+    operatorState.cachedLogger(`Checking eligibility for ${nodeLicenseIds.length} Keys.`);
+
+    let nonWinnerKeysCount = 0;
+
+    for (const nodeLicenseId of nodeLicenseIds) {
+
+        const sentryKey = sentryKeysMap[nodeLicenseId.toString()];
+
+        // Check the nodeLicense eligibility for this challenge
+        updateNodeLicenseStatus(nodeLicenseId, NodeLicenseStatus.CHECKING_MINT_TIMESTAMP_ELIGIBILITY);
+
+        if (challenge.createdTimestamp <= sentryKey.mintTimeStamp) {
+            operatorState.cachedLogger(`Sentry Key ${nodeLicenseId} is not eligible for challenge ${challengeNumber}.`);
+            updateNodeLicenseStatus(nodeLicenseId, NodeLicenseStatus.WAITING_FOR_NEXT_CHALLENGE);
+            continue;
+        }
+
+        // Check if nodeLicense is eligible for a payout
+        updateNodeLicenseStatus(nodeLicenseId, NodeLicenseStatus.CHECKING_IF_ELIGIBLE_FOR_PAYOUT);
+
+        try {
+            let isPool = sentryKey.assignedPool != "0x";
+            const keyOwner = isPool ? sentryKey.assignedPool : sentryKey.owner;
+            if (!operatorState.cachedBoostFactor[keyOwner]) {
```

Comment:
If a pool's boost factor changes, will this cached boost factor ever be "refreshed"?

Reply:
Currently at every challenge it will be recalculated with `operatorState.cachedBoostFactor = {};`